src/util/strencodings.h

```diff
@@ -54,7 +54,7 @@ enum class ByteUnit : uint64_t {
 * @return           A new string without unsafe chars
 */
 std::string SanitizeString(const std::string& str, int rule = SAFE_CHARS_DEFAULT);
-std::vector<unsigned char> ParseHex(const char* psz);
+/** Parse the hex string into bytes. Ignores whitespace. */
 std::vector<unsigned char> ParseHex(const std::string& str);
```

Comment:
This is a pessimization if the passed-in parameter is actually a `const char *`.

Reply:
There are a few places that use the pointer interface. However, they are not performance critical:

* Unit tests
* Calls, that are only done once (genesis block, other chain params, ...)
* In net processing to generate the alert message

I think for none of these performance matter, because they are either (1) small strings, in which case a copy doesn't matter, (2) used in tests or otherwise places where performance isn't needed.

The places that are affected (RPC mostly), already use `std::string` to pass around hex stuff. For them this change has no effect, but likely #https://github.com/bitcoin/bitcoin/pull/18061 helps here. For example, when hex-decoding a 1-MB block or so.

Reply:
I think switching to `string_view` could still make sense, but this requires a re-write of the parse-algorithm, as string view lacks the trailing null-char, which the current algorithm relies on. (There is no `std::string_view::c_str()` method)

Reply:
OK. I still think being able to pass in a memory area instead of a full, allocated string is a conceptually useful interface. `const char*` is just the wrong way to do it.

Reply:
> I think switching to string_view could still make sense, but this requires a re-write of the parse-algorithm,

Fine, yes that may be too much for a refactor like this, let's not do it in this PR then.

Reply:
(unresolved conversation, to avoid hiding it)